File: cpp/program5.cpp
// program87.c

#include<iostream>
using namespace std;

class Array
{
    public:
        int iSize;
        int *Arr;

     Array(int iNo1)
    {
        iSize = iNo1;
        Arr = new int[iSize];
    }

    void Accept()
    {
        cout<<"Enter the elements"<<'\n';

        int iCnt = 0;
        for(iCnt = 0; iCnt < iSize; iCnt++)
        {
            cin>>Arr[iCnt];
        }
    }

    ~Array()
    {
        delete []Arr;
    }

    bool SearchNumber()
    {
        int iNo2 = 0;
        bool bFlag = false;

        cout<<"Enter the number you want to search"<<"\n";
        cin>>iNo2;

        int iCnt = 0;
        for(iCnt = 0; iCnt < iSize; iCnt++)
        {
           if(Arr[iCnt] == iNo2)
           {
                bFlag = true;
                break;
           }
        }
        return bFlag;
    }
};

int main()
{
    int iValue = 0;
    bool bRet = false;
    
    cout<<"Enter number of elements"<<"\n";
    cin>>iValue;

    Array aobj(iValue);
    aobj.Accept();
    bRet = aobj.SearchNumber();

    if(bRet == false)
    {
        cout<<"There is no such number"<<"\n";
    }
    else
    {
        cout<<"Number is present "<<"\n";
    }

    return 0;
}

File: scala/program5.scala
import scala.io.StdIn.{readInt};

class ArrayX(var size : Int) 
{
    val arr = new Array[Int](size);

    def Accepts() {
        println("Enter elements of the array : ")
        for(i <- 0 until size)
        {
            arr(i) = readInt();
        }
    }

    def Display() {
        println("Elements of the array are :")
        
        for(num <- arr) {
            println(num)
        }
    }

    def Search(target : Int): Boolean = {
        for(num <- arr) {
            if(num == target) return true;
        }
        false;
    }
}

object Demo
{
    def main(args : Array[String]) 
    {
        println("Enter number of elements : ");
        var size : Int = readInt();

        var obj = new ArrayX(size);
        obj.Accepts();
        obj.Display();

        println("Enter elements to check in the array : ")
        var target : Int = readInt();

        var bRet = obj.Search(target)
        if(bRet == true) { 
            println("Element is found")
        } else {
            println("Element not present");
        }
    }
}